agent.go

```diff
@@ -224,6 +226,12 @@ func (a Agent) GetRootKey() []byte {
 	return a.rootKey
 }
 
+func newNonce() []byte {
+	nonce := make([]byte, 10)
```

Comment:
Is there a reason we use a `10` byte nonce? This is probably enough to differentiate the same request within the same time period.

> nonce (blob, optional): Arbitrary user-provided data of length at most 32 bytes, typically randomly generated. This can be used to create distinct requests with otherwise identical fields.

Reply:
> This is probably enough to differentiate the same request within the same time period.

That's what I thought, and I thought 32 bytes would be overkill. I'll add a note about that "choice"!

Reply:
@q-uint done! merge at will